pkg/sync/adapter.go

```diff
@@ -0,0 +1,222 @@
+package sync
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/devfile/adapters/common"
+	versionsCommon "github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/exec"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/util"
+
+	"github.com/pkg/errors"
+)
+
+// New instantiantes a component adapter
+func New(adapterContext common.AdapterContext, client SyncClient) Adapter {
+	return Adapter{
+		Client:         client,
+		AdapterContext: adapterContext,
+	}
+}
+
+// Adapter is a component adapter implementation for sync
+type Adapter struct {
+	Client SyncClient
+	common.AdapterContext
+}
+
+// CheckProjectFiles checks whether files have changed in a project
+func (a Adapter) CheckProjectFiles(parameters common.PushParameters, podName, containerName string, podChanged, componentExists bool) (err error) {
+
+	deletedFiles := []string{}
+	changedFiles := []string{}
+	isForcePush := false
+
+	globExps := util.GetAbsGlobExps(parameters.Path, parameters.IgnoredFiles)
+
+	// Sync source code to the component
+	// If syncing for the first time, sync the entire source directory
+	// If syncing to an already running component, sync the deltas
+	// If syncing from an odo watch process, skip this step, as we already have the list of changed and deleted files.
+	if !podChanged && !parameters.ForceBuild && len(parameters.WatchFiles) == 0 && len(parameters.WatchDeletedFiles) == 0 {
+		absIgnoreRules := util.GetAbsGlobExps(parameters.Path, parameters.IgnoredFiles)
+
+		spinner := log.NewStatus(log.GetStdout())
```

Comment:
Can you use `log.Spinner()` for the spinner here? To be consistent with the rest of odo. And pair that with a `defer s.End(false)`